src/engine/passwordless.js

```diff
@@ -16,6 +16,19 @@ import { isDone, isSuccess } from '../sync';
 import * as l from '../core/index';
 import { hasSkippedQuickAuth } from '../quick_auth';
 import * as sso from '../core/sso/index';
+import { setEmail } from '../field/email';
+import { setPhoneNumber } from '../field/phone_number';
+
+const setPrefill = m => {
+  const { email, phoneNumber } = l.prefill(m).toJS();
+  if (typeof email === 'string') {
+    m = setEmail(m, email);
+  }
+  if (typeof phoneNumber === 'string') {
+    m = setPhoneNumber(m, phoneNumber);
```

Comment:
the previous `m` value get's overwritten with this one if both if clauses are called. Is this intended? what means the `m` returned?

Reply:
yeah, that's how lock's internal state work. `m` is an immutable object, which means that each `setSOMETHING` method returns a new instance of the model